File: frontend/types/meeting.ts
export type MeetingStatus = "upcoming" | "completed" | "cancelled";

export interface Meeting {
  id: number;
  company_id: number;
  meeting_date: string;
  location: string | null;
  notes: string | null;
  status: MeetingStatus;
  created_at: string;
}

export interface MeetingCreate {
  company_id: number;
  meeting_date: string;
  location?: string;
  notes?: string;
  status?: MeetingStatus;
}

export interface Meeting {
  id: number;
  company_id: number;
  company_name: string;   // added
  meeting_date: string;
  location: string | null;
  notes: string | null;
  status: MeetingStatus;
  created_at: string;
}

File: backend/app/schemas/meeting.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional
from app.models.meeting import MeetingStatus


class MeetingBase(BaseModel):
    company_id: int
    meeting_date: datetime
    location: Optional[str] = None
    notes: Optional[str] = None
    status: MeetingStatus = MeetingStatus.UPCOMING


class MeetingCreate(MeetingBase):
    pass


class MeetingOut(MeetingBase):
    id: int
    created_at: datetime
    company_name: str   # added: resolved from the related Company

    class Config:
        from_attributes = True